c2corg_ui/static/js/card.js

```diff
@@ -61,17 +60,23 @@ app.CardController = function(gettextCatalog) {
 
   /**
    * @type {Object}
-   * @private
+   * @export
+   */
+  this.doc;
+
+  /**
+   * @type {string}
+   * @export
    */
-  this.doc_ = this['doc'];
+  this.type = app.utils.getDoctype(this.doc['type']);
 
   /**
    * @type {Object}
    * @export
    */
-  this.locale = this['doc']['locales'][0];
-  for (var i = 0, n = this['doc']['locales'].length; i < n; i++) {
-    var l = this['doc']['locales'][i];
+  this.locale = this.doc['locales'][0];
```

Comment:
It seems to work indeed. But I am not sure to understand why because I don't really see the typedef for documents in the externs https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/externs/appx.js